File: forgotPassForm.cs
using System;
using System.Net;
using System.Net.Mail;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace MilkTeaOrderAndInventorySystem
{

    public partial class forgotPassForm
    {
        private string randomCode;
        public bool passedValue { get; set; } = false;

        public static string toUser;

        public forgotPassForm()
        {
            InitializeComponent();
        }

        private void Guna2Button2_Click(object sender, EventArgs e)
        {
            var obj = new Login();
            obj.Show();
            Hide();
        }

        private void createNewAcc_Click(object sender, EventArgs e)
        {
            var obj = new Login();
            obj.signupPanel.Visible = true;
            obj.loginPanel.Visible = false;
            obj.Show();
            Hide();
        }

        private void Guna2TextBox1_TextChanged(object sender, EventArgs e)
        {
            txtError.Visible = false;

            txtError.Visible = false;
            if (string.IsNullOrEmpty(emailReset.Text))
            {
                btnSend.Enabled = false;
            }
            else
            {
                btnSend.Enabled = true;
            }
        }

        private void btnSend_Click(object sender, EventArgs e)
        {
            sendCode();
        }

        private void sendCode()
        {
            if (IsValidEmailFormat(emailReset.Text))
            {
                emailCheck();
            }
            else
            {
                txtError.Visible = true;
                txtError.Text = "Please enter a valid email address!";
            }
        }

        private void Guna2Button3_Click(object sender, EventArgs e)
        {
            verifyCode();
        }

        private void verifyCode()
        {
            if (passedValue == true)
            {
                if (txtCode.Text.Equals(randomCode))
                {
                    var obj = new ResetPass()
                    {
                        passEmail = emailReset.Text,
                        passValue = passedValue
                    };
                    obj.btnLogin.Text = "Done";
                    obj.Show();
                    Hide();
                }
                else
                {
                    txtCodeCheck.Visible = true;
                    txtCodeCheck.Text = "Invalid Code!";
                }
            }
            else if (txtCode.Text.Equals(randomCode))
            {
                var obj = new ResetPass() { passEmail = emailReset.Text };
                obj.Show();
                Hide();
            }
            else
            {
                txtCodeCheck.Visible = true;
                txtCodeCheck.Text = "Invalid Code!";
            }
        }

        public bool IsValidEmailFormat(string s)
        {
            try
            {
                var a = new MailAddress(s);
            }
            catch
            {
                return false;
            }
            return true;
        }

        private void emailCheck()
        {
            UniversalVariables.sql = "SELECT * FROM userProfile WHERE uEmail = '" + emailReset.Text + "'";
            UniversalVariables.connect();

            if (UniversalVariables.dr.Read())
            {
                string from, pass, messageBody;
                var rand = new Random();
                randomCode = rand.Next(999999).ToString();
                var message = new MailMessage();
                toUser = emailReset.Text;
                from = "trialkoito4@gmail.com";
                pass = "broxxedasqosbrvy";
                messageBody = "Your Verification Code is: " + randomCode;
                message.To.Add(toUser);
                message.From = new MailAddress(from);
                message.Body = messageBody;
                message.Subject = "Resetting Code";
                var smtp = new SmtpClient("smtp.gmail.com")
                {
                    EnableSsl = true,
                    Port = 587
                };
                smtp.DeliveryMethod = SmtpDeliveryMethod.Network;
                smtp.Credentials = new NetworkCredential(from, pass);
                try
                {
                    smtp.Send(message);

                    var obj1 = new CodeVerification();
                    obj1.ShowDialog();

                    sendPanel.Hide();
                    verifyPanel.Show();
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message);
                }
            }
            else
            {
                txtError.Visible = true;
                txtError.Text = "Email is not found!";
            }
            UniversalVariables.conn.Close();
        }

        private void forgotPassForm_Load(object sender, EventArgs e)
        {
            UniversalVariables.conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb";
            txtCode.MaxLength = 6;
        }

        private void Guna2TextBox2_TextChanged(object sender, EventArgs e)
        {
            txtCodeCheck.Visible = false;

            if (string.IsNullOrEmpty(txtCode.Text))
            {
                btnVerify.Enabled = false;
            }
            else
            {
                btnVerify.Enabled = true;
            }
        }

        private void txtCode_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsNumber(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.KeyChar = Conversions.ToChar("");
        }

        private void emailReset_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                sendCode();
            }
        }

        private void txtCode_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                verifyCode();
            }
        }
    }
}

File: forgotPassForm.vb
Imports System.Net
Imports System.Net.Mail

Public Class forgotPassForm
    Dim randomCode As String
    Public Property passedValue As Boolean = False

    Public Shared toUser As String

    Private Sub Guna2Button2_Click(sender As Object, e As EventArgs) Handles btnBack.Click
        Dim obj As New Login
        obj.Show()
        Me.Hide()
    End Sub

    Private Sub createNewAcc_Click(sender As Object, e As EventArgs) Handles createNewAcc.Click
        Dim obj As New Login
        obj.signupPanel.Visible = True
        obj.loginPanel.Visible = False
        obj.Show()
        Me.Hide()
    End Sub

    Private Sub Guna2TextBox1_TextChanged(sender As Object, e As EventArgs) Handles emailReset.TextChanged
        txtError.Visible = False

        txtError.Visible = False
        If emailReset.Text = Nothing Then
            btnSend.Enabled = False
        Else
            btnSend.Enabled = True
        End If
    End Sub

    Private Sub btnSend_Click(sender As Object, e As EventArgs) Handles btnSend.Click
        sendCode()
    End Sub

    Private Sub sendCode()
        If IsValidEmailFormat(emailReset.Text) Then
            emailCheck()
        Else
            txtError.Visible = True
            txtError.Text = "Please enter a valid email address!"
        End If
    End Sub

    Private Sub Guna2Button3_Click(sender As Object, e As EventArgs) Handles btnVerify.Click
        verifyCode()
    End Sub

    Private Sub verifyCode()
        If passedValue = True Then
            If (txtCode.Text.Equals(randomCode)) Then
                Dim obj As New ResetPass With {
                    .passEmail = emailReset.Text,
                    .passValue = passedValue
                }
                obj.btnLogin.Text = "Done"
                obj.Show()
                Me.Hide()
            Else
                txtCodeCheck.Visible = True
                txtCodeCheck.Text = "Invalid Code!"
            End If
        Else
            If (txtCode.Text.Equals(randomCode)) Then
                Dim obj As New ResetPass With {
                    .passEmail = emailReset.Text
                }
                obj.Show()
                Me.Hide()
            Else
                txtCodeCheck.Visible = True
                txtCodeCheck.Text = "Invalid Code!"
            End If
        End If
    End Sub

    Public Function IsValidEmailFormat(ByVal s As String) As Boolean
        Try
            Dim a As New System.Net.Mail.MailAddress(s)
        Catch
            Return False
        End Try
        Return True
    End Function

    Private Sub emailCheck()
        sql = "SELECT * FROM userProfile WHERE uEmail = '" & emailReset.Text & "'"
        connect()

        If dr.Read Then
            Dim from, pass, messageBody As String
            Dim rand As New Random()
            randomCode = (rand.Next(999999)).ToString()
            Dim message As New MailMessage()
            toUser = emailReset.Text
            from = "trialkoito4@gmail.com"
            pass = "broxxedasqosbrvy"
            messageBody = "Your Verification Code is: " + randomCode
            message.To.Add(toUser)
            message.From = New MailAddress(from)
            message.Body = messageBody
            message.Subject = "Resetting Code"
            Dim smtp As New SmtpClient("smtp.gmail.com") With {
                .EnableSsl = True,
                .Port = 587
            }
            smtp.DeliveryMethod = smtp.DeliveryMethod.Network
            smtp.Credentials = New NetworkCredential(from, pass)
            Try
                smtp.Send(message)

                Dim obj1 As New CodeVerification
                obj1.ShowDialog()

                sendPanel.Hide()
                verifyPanel.Show()
            Catch ex As Exception
                MessageBox.Show(ex.Message)
            End Try
        Else
            txtError.Visible = True
            txtError.Text = "Email is not found!"
        End If
        conn.Close()
    End Sub

    Private Sub forgotPassForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb"
        txtCode.MaxLength = 6
    End Sub

    Private Sub Guna2TextBox2_TextChanged(sender As Object, e As EventArgs) Handles txtCode.TextChanged
        txtCodeCheck.Visible = False

        If txtCode.Text = Nothing Then
            btnVerify.Enabled = False
        Else
            btnVerify.Enabled = True
        End If
    End Sub

    Private Sub txtCode_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtCode.KeyPress
        If Not Char.IsNumber(e.KeyChar) AndAlso Not Char.IsControl(e.KeyChar) Then e.KeyChar = ""
    End Sub

    Private Sub emailReset_KeyDown(sender As Object, e As KeyEventArgs) Handles emailReset.KeyDown
        If e.KeyCode = Keys.Enter Then
            sendCode()
        End If
    End Sub

    Private Sub txtCode_KeyDown(sender As Object, e As KeyEventArgs) Handles txtCode.KeyDown
        If e.KeyCode = Keys.Enter Then
            verifyCode()
        End If
    End Sub
End Class